### ISSUE TITLE
Information about the used train-val-test split

### ISSUE THREAD

**Issue by lmanan** [CAREamics/MicroSplit-reproducibility#21]
Hi, 

The [links to all datasets](https://github.com/CAREamics/MicroSplit-reproducibility#links-to-all-datasets-used-in-the-manuscript) in the manuscript do not have a train-val-test split, could you provide info about which of the files were used for training, validation and evaluation, please?
Also, which 3 of the 6 channels were used in the HHMI-D25 dataset? I couldn't locate this information in the manuscript.
Finally, the links in the example notebooks such as https://download.fht.org/jug/msplit/ht_lif24/data/ seem to not work anymore for me. It directs me to a 403 forbidden page. 

Thank you!

**Comment by jdeschamps** [comment]
Hi @lmanan !

Thanks for picking up on that.

It is a bank holiday today, we will get back to you early next week.

**Comment by jdeschamps** [comment]
Hi Manan,

Very sorry for the delay.

Just a note: this repository reproduces some results of the paper, most experiments were run [juglab/MicroSplit](https://github.com/juglab/MicroSplit/). The reason being that the paper was in review for a while and in the mean time we did some heavy refactoring of the code (this repo).

> The [links to all datasets](https://github.com/CAREamics/MicroSplit-reproducibility#links-to-all-datasets-used-in-the-manuscript) in the manuscript do not have a train-val-test split, could you provide info about which of the files were used for training, validation and evaluation, please?

There is no easy way to give the exact train/val/test split without running the original code and extracting the patches. A seed is fixed, ensuring the exact same split every time:
- https://github.com/juglab/MicroSplit/blob/master/disentangle/core/data_split_type.py#L38, this is called in many dataloaders
- https://github.com/juglab/MicroSplit/blob/master/disentangle/data_loader/elisa3D_rawdata_loader.py#L35

Note that we currently wrap the original datasets and utility functions in CAREamics/reproducibility, meaning that it should yield the same result. All this will be soon refactored to be cleaner.

Would you want us to extract precise patch / source coordinates for the splits? What would be your purpose?

@ashesh-0 can you chime in?

> Also, which 3 of the 6 channels were used in the HHMI-D25 dataset? I couldn't locate this information in the manuscript.

It may have been the channels indicated here:
- https://github.com/juglab/MicroSplit/blob/cbdcb93421d05000b5efa43d5922341274102959/disentangle/data_loader/hhmi_v2_raw_dloader.py#L59

But for this we need to ask @ashesh-0 directly.

> Finally, the links in the example notebooks such as https://download.fht.org/jug/msplit/ht_lif24/data/ seem to not work anymore for me. It directs me to a 403 forbidden page.

I do not have any issue (even outside of HT). Could you try again (maybe I changed the permissions while checking what may be the problem)?

Here is a direct example: https://download.fht.org/jug/msplit/ht_lif24/data/ht_lif24_5ms.zip

**Comment by ashesh-0** [comment]
Hi @lmanan ,
We have worked with 36 tasks, meaning 36 train/val/test from 9 data sources, which vary in many aspects. For instance, some sources have a large spatial dimension but few files, while others have more files containing smaller-sized z-stacks. So, the logic for train/test/val varies. You can find that logic here https://github.com/juglab/MicroSplit/blob/master/disentangle/data_loader/train_val_data.py. 

At this point, as @jdeschamps mentioned, there is no simple way to get the train/val/test data, something that any person wanting to build upon or test MicroSplit would like to do. 

At this point, what might be useful is to look at a training notebook: https://github.com/CAREamics/MicroSplit-reproducibility/blob/main/examples/2D/HT_LIF24/01_train.ipynb

All you need to do is copy the first few cells till the point of 
```
train_dset, val_dset, _, data_stats = create_train_val_datasets(
    datapath=DATA.path / f"ht_lif24_{EXPOSURE_DURATION}.zip.unzip/{EXPOSURE_DURATION}",
    train_config=train_data_config,
    val_config=val_data_config,
    test_config=val_data_config,
    load_data_func=get_train_val_data,
)
```
You will need to make several changes. 
- In the test_config argument, put the `test_data_config`. 
- You will have to update the `datapath` for each dataset as well. 
- You will also need to import dataset-specific functions (see below).
 
```
# Dataset specific imports...
from microsplit_reproducibility.configs.parameters.HT_LIF24 import get_microsplit_parameters
from microsplit_reproducibility.configs.data.HT_LIF24 import get_data_configs
from microsplit_reproducibility.datasets.HT_LIF24 import get_train_val_data
```

Regarding the HHMI channels, @jdeschamps is right. They are `[1,2,4]` (zero-indexed). 
Hope that helps.

**Comment by lmanan** [comment]
Thank you Joran @jdeschamps and Ashesh @ashesh-0 
Understood that the train-val-test split is dependent on the task. Thank you for the detailed replies! 
This link (https://download.fht.org/jug/msplit/ht_lif24/data/) still doesn't work for me from US mainland, gives me a 403 Forbidden error, I am not sure why. But it is not critical, for me, I am happy to close this issue for now. Thank you both!

**Comment by jdeschamps** [comment]
> Thank you Joran [@jdeschamps](https://github.com/jdeschamps) and Ashesh [@ashesh-0](https://github.com/ashesh-0) Understood that the train-val-test split is dependent on the task. Thank you for the detailed replies! This link (https://download.fht.org/jug/msplit/ht_lif24/data/) still doesn't work for me from US mainland, gives me a 403 Forbidden error, I am not sure why. But it is not critical, for me, I am happy to close this issue for now. Thank you both!

That specific look should not work. If you look at the notebook, you'll see that the link is generated as a f-string using the exposure time to build the name.

The example I gave you is for `5ms`:
https://download.fht.org/jug/msplit/ht_lif24/data/ht_lif24_5ms.zip

The available exposures are:
```
VeryLow = "2ms"
Low = "3ms"
Medium = "5ms"
High = "20ms"
VeryHigh = "500ms"
```

So:
https://download.fht.org/jug/msplit/ht_lif24/data/ht_lif24_2ms.zip
https://download.fht.org/jug/msplit/ht_lif24/data/ht_lif24_3ms.zip
https://download.fht.org/jug/msplit/ht_lif24/data/ht_lif24_5ms.zip
https://download.fht.org/jug/msplit/ht_lif24/data/ht_lif24_20ms.zip
https://download.fht.org/jug/msplit/ht_lif24/data/ht_lif24_500ms.zip

**Comment by lmanan** [comment]
Ah I see, I should have noticed that the url was being concatenated with another string, that's my bad. Thank you Joran!

**Comment by jdeschamps** [comment]
No worries! Let us know if you stumble on something else!
